File: src/BrokerageExportProviders/Brokerages/IBKR/Schemas/SegmentedTrades.py
import BrokerageExportProviders.Brokerages.IBKR.Schemas.Schemas as s
import BrokerageExportProviders.Contracts.CommonBrokerageEvents as cbe


class SegmentedTrades(
    cbe.CommonBrokerageEvents[s.TransactionCash, s.CorporateAction, s.TradeStock, s.LotStock, s.TradeDerivative, s.LotDerivative]
): ...

File: lib/Brokerages/Ibkr/Schemas/SegmentedTrades.ts
import type { CorporateAction, LotDerivative, LotStock, TradeDerivative, TradeStock, TransactionCash } from "@brrr/Brokerages/Ibkr/Schemas/IbkrSchemas.ts";

export type SegmentedTrades = {
	cashTransactions: TransactionCash[];
	corporateActions: CorporateAction[];
	stockTrades: TradeStock[];
	stockLots: LotStock[];
	derivativeTrades: TradeDerivative[];
	derivativeLots: LotDerivative[];
};
